### PR TITLE
trim cron string

### PR DESCRIPTION
**Pull request by pnaik1**
https://issues.redhat.com/browse/RHOAIENG-9537

## Description
the user can end up with an error while submitting a scheduled run because of white space to avoid this I have trim white spaces from cron string

## How Has This Been Tested?

1. Create a scheduled run
2. Change the Trigger type to cron
3. at @every 5m with a space at the end
4. check if the scheduled run is created successfully without any error

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/pipelines/content/createRun/contentSections/TriggerTypeField.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ const TriggerTypeField: React.FC<TriggerTypeFieldProps> = ({ data, onChange }) =
             clickTip="Copied"
             onChange={(e, value) => {
               if (typeof value === 'string') {
-                onChange({ ...data, value });
+                onChange({ ...data, value: value.trim() });
```
Comment (diegolovison) [inline frontend/src/concepts/pipelines/content/createRun/contentSections/TriggerTypeField.tsx:39]: Hi, why we don't have a test for this?

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineCreateRuns.cy.ts [hunk 2]
```diff
@@ -590,6 +590,8 @@ describe('Pipeline create runs', () => {
       createSchedulePage.findExperimentSelect().should('contain.text', 'Default');
       createSchedulePage.findExperimentSelect().should('not.be.disabled').click();
       createSchedulePage.selectExperimentByName('Test experiment 1');
+      createSchedulePage.findScheduledRunTypeSelector().findDropdownItem('Cron').click();
+      createSchedulePage.findScheduledRunCron().fill('@every 5m ');
```
Comment (diegolovison) [inline frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineCreateRuns.cy.ts:594]: Where do you assert for `@every 5m`? Without the space in the end?

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineCreateRuns.cy.ts [hunk 3]
```diff
@@ -630,6 +621,36 @@ describe('Pipeline create runs', () => {
...
+          display_name: 'New recurring run',
+          description: 'New recurring run description',
+          pipeline_version_reference: {
+            pipeline_id: 'test-pipeline',
+            pipeline_version_id: 'test-pipeline-version',
+          },
+          runtime_config: {
+            parameters: { min_max_scaler: false, neighbors: 1, standard_scaler: 'no' },
+          },
+          trigger: { cron_schedule: { cron: '@every 5m' } },
```
Comment (diegolovison) [inline frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineCreateRuns.cy.ts:644]: Where is the test for '@every 5m '
The one that contains the space?

Comment (pnaik1) [reply frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelineCreateRuns.cy.ts:644]: @diegolovison , I have created two separate case 
the one with whitespace  is in this line https://github.com/opendatahub-io/odh-dashboard/pull/3061/files/3d1bd4b1eb2198f3dca928339229211e5bb11cf0#diff-ff1a31b964e7adebb99b1bee1eaa6d2678fd21025df88198dd7da5ade83aa8aaR627

**Review by DaoDaoNoCode — APPROVED** [review]
/lgtm

**Pull request merged**